objects/src/accounts/vault.rs

```diff
@@ -204,6 +204,34 @@ impl AccountVault {
     }
 }
 
+// SERIALIZATION
+// ================================================================================================
+
+impl Serializable for AccountVault {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        // TODO: determine total number of assets in the vault without allocating the vector
+        let assets = self.assets().collect::<Vec<_>>();
+
+        assert!(assets.len() <= u32::MAX as usize, "too many assets in the vault");
```

Comment:
Good point! I'll add a TODO to address this later.

With Winterfell's serialization approach, the serialize assumes that an object is in a valid state - so, the way to fix it would be to check the total number of assets in the constructor and make sure it fits into `u32::Max`.

Reply:
Actually, probably a better solution would be to serialize `usize`'s as variable-length integers. This would need to be one in Winterfell though as right now there is no way to serialize/deserialize `usize` values there.